float32[] coordinate
float32 theta
float32 vel_b
float32 phi
float32 ctrl_vel_b
float32 ctrl_phi
bool movable
bool collision